File: robocode2-bot-api-java/src/main/java/net/robocode2/events/BulletHitBulletEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Value;
import net.robocode2.BulletState;

/** Event occurring when a bullet has collided with another bullet */
@Value
@EqualsAndHashCode(callSuper = true)
public class BulletHitBulletEvent extends GameEvent {
  /** Bullet that hit another bullet */
  BulletState bullet;
  /** The other bullet that was hit by the bullet */
  BulletState hitBullet;

  @Builder
  private BulletHitBulletEvent(int turnNumber, BulletState bullet, BulletState hitBullet) {
    this.turnNumber = turnNumber;
    this.bullet = bullet;
    this.hitBullet = hitBullet;
  }
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/BulletHitBulletEvent.kt
package net.robocode2.gui.model

data class BulletHitBulletEvent(
        override val turnNumber: Int,
        val bullet: BulletState,
        val hitBullet: BulletState
) : Event(MessageType.BULLET_HIT_BULLET_EVENT.type, turnNumber)
